### ISSUE TITLE
Improvements for headless use

### ISSUE THREAD

**Issue by schnatterer** [marcosnils/bin#102]
It would be great to have a non-interactive mode where bin does not prompt.

I'm starting to role out bin to all my machines headlessly, using ansible ([bin installation, self management and update](https://github.com/schnatterer/devboxes/blob/main/roles/cli-tools/tasks/bin.yml); [bin install example](https://github.com/schnatterer/devboxes/blob/bcae886b3ff720f250282fc94fbdfba6d48ad57e/roles/cli-tools/tasks/gotop.yml)).

Here it is difficult to install some binaries because of the prompt. This could be worked around with, e.g. `echo 4 | bin install ....`.
But update all managed binaries it is impossible because of different prompts.

So basically I'd love to have two features:
* Choosing a binary or binaries via a parameter (that can be passed multiple times for multiple binaries), that evaluates a regex
* Storing this parameter in `config.json`, so it can be evaluated again to make headless `udpate`s possible.

Examples:
* [github.com/ahmetb/kubectx](https://github.com/ahmetb/kubectx) - choose `kubectx` and `kubens`
* [github.com/ogham/exa](https://github.com/ogham/exa) - choose exa without `musl`
* [github.com/cloudflare/cfssl](https://github.com/cloudflare/cfssl) - choose multiple binaries

**Comment by marcosnils** [comment]
> * Choosing a binary or binaries via a parameter (that can be passed multiple times for multiple binaries), that evaluates a regex

Yep, we've briefly spoken about this in some other issues. I think this could be a good place to start prototyping and define how the headless UX would look like. 

@schnatterer would you like to kick this off?

**Comment by skorokithakis** [comment]
I'm not sure if this is the right issue, but there are some very simple things that could be done now (e.g. ignore files with `sha256` in the extension). Could some of these improvements be implemented?

**Comment by marcosnils** [comment]
> I'm not sure if this is the right issue, but there are some very simple things that could be done now (e.g. ignore files with `sha256` in the extension). Could some of these improvements be implemented?

yep. definitely, I'll have some time during the next days to check this out.

**Comment by skorokithakis** [comment]
Thank you! I imagine there are a few more things that can be done to weed out various candidates, e.g. man pages, or txt files, etc. Unfortunately, I just threw out my output after upgrading, but I can post some easy pickings next time I do.

Presumably, `bin` removes those files from consideration and then installs what's left (or lets you choose what to install, if there are more than one candidates left).

**Comment by marcosnils** [comment]
Yep, the initial approach would be to `only` list executable files initially and see how it goes from there.

**Comment by giggio** [comment]
I found that if you create the config file it will not prompt you to select the install location:

```bash
# run as root
echo '{ "default_path": "/usr/local/bin", "bins": { } }' > "$HOME"/.config/bin/config.json
```
